scripts/test.mk

```diff
@@ -0,0 +1,52 @@
+# This file is part of the repository template project. Please, DO NOT edit this file.
+
+# The test types listed here are both those which might run both locally and in CI, or
+# in one but not the other.  All of the test types listed at
+# https://github.com/NHSDigital/software-engineering-quality-framework/blob/main/quality-checks.md
+# should be represented here with the exception of:
+#   - dependency scanning, which we expect to be applied at the repository level
+#   - secret scanning, which we expect to be a pre-commit hook
+#   - code review, which is outside the scope of automated testing for the moment
+
+test-integration:
+	@make _test name="integration"
+
+test-contract:
+	@make _test name="contract"
+
+test-ui:
+	@make _test name="ui"
+
+test-accessibility:
+	@make _test name="accessibility"
+
+test-lint:
+	@make _test name="lint"
+
+# test-code-quality covers checking for duplicate code, code smells, and dead code.
```

Comment:
Hm.  I'm not sure if that's right. The comment is to explain the disparity between the tests listed in `test.mk` and the test types listed in the quality framework, which are otherwise a one-to-one match. That's only something you'd really care about if you're working on the template itself. It's not really focused on the point of use.

What I've done is to add user-facing comments to each test task so they appear in the `help` output.  I'm not wild about them though - happy to take suggestion if there's something that reads better.

```
 $ make help
_install-dependency                       Install asdf dependency - mandatory: name=[listed in the `./.tool-versions` file]; optional: version=[if not listed]
asdf-install                              Install asdf from https://asdf-vm.com/
clean                                     Remove all generated and temporary files
config                                    Configure development environment
githooks-install                          Install git hooks configured in this repository
githooks-run                              Run git hooks configured in this repository
help                                      List Makefile targets
list-variables                            List all the variables available to make
nodejs-install                            Install Node.js
python-install                            Install Python
terraform-install                         Install Terraform
test                                      Run all the test tasks
test-accessibility                        Run your accessibility tests from scripts/test/accessibility
test-code-quality                         Run your code quality tests from scripts/test/code-quality
test-contract                             Run your contract tests from scripts/test/contract
test-coverage                             Evaluate code coverage from scripts/test/coverage
test-integration                          Run your integration tests from scripts/test/integration
test-lint                                 Lint your code from scripts/test/lint
test-ui                                   Run your UI tests from scripts/test/ui
test-unit                                 Run your unit tests from scripts/test/unit
```

Reply:
Thanks @regularfry, it is a valid comment to keep as a guide.